packages/token-vault/src/lib/proxy.ts

```diff
@@ -1,60 +1,101 @@
+import {
+  cloneResponse,
+  createErrorResponse,
+  generateAmUrls,
+} from '@shared/network';
+import { EventsConfig, ProxyConfig, ServerTokens } from '@shared/types';
 import {
   refreshTokens,
   storeTokens,
-  requestTokens,
   getTokens,
-} from './utils/token';
-import {
-  cloneResponse,
-  createErrorResponse,
-  generateUrls,
-} from '@forgerock/network-utils';
-import type {
-  ProxyConfig,
-  ResponseClone,
-  ServerTokens,
-} from '@forgerock/shared-types';
-
+  tokenExpiryWithinThreshold,
+} from './token.utils';
+
+/** ****************************************************************
+ * @function proxy - Initialize the Token Vault Proxy
+ * @param {ProxyConfig} config - The configuration object for the proxy
+ * @returns {void}
+ * @example proxy({
+ *   forgerock: { ... },
+ *   app: { ... },
+ *   proxy: { ... }
+ * });
+ */
 export function proxy(config: ProxyConfig) {
   /**
-   * Client configuration
+   * Client default configuration
    */
   const clientId = config.forgerock?.clientId || 'WebOAuthClient';
   const clientOrigin = config.app.origin || 'http://localhost:8000';
+  const oauthThreshold = config.forgerock?.oauthThreshold || 30 * 1000;
   const scope = config.forgerock?.scope || 'openid email';
+  const redactedTokens = config.proxy?.redact || [
+    'access_token',
+    'refresh_token',
+  ];
+
+  /**
+   * Default event names
+   * These can be overridden in the config
+   */
+  const eventDefaults = [
+    { name: 'TVP_FETCH_RESOURCE', type: 'fetch' },
+    { name: 'TVP_HAS_TOKENS', type: 'has' },
+    { name: 'TVP_REFRESH_TOKENS', type: 'refresh' },
+    { name: 'TVP_REMOVE_TOKENS', type: 'remove' },
+  ] as const;
 
   /**
-   * Event names
+   * Generate the event names provided in the config or use the defaults
    */
-  const fetchEventName = config?.events?.fetch || 'FETCH_RESOURCE';
-  const hasTokenEventName = config?.events?.has || 'HAS_TOKENS';
-  const refreshTokenEventName = config?.events?.refresh || 'REFRESH_TOKENS';
-  const removeTokenEventName = config?.events?.remove || 'REMOVE_TOKENS';
+  const eventsObj = eventDefaults.reduce((acc, event) => {
+    const eventName = config.events?.[event.type] || event.name;
+    acc[event.type] = eventName;
+    return acc;
+  }, {} as EventsConfig);
```

Comment:
This is kind of a fancy way to generate the event objects for the conditionals below with the capability to customize them by the dev, if they wanted.